controllers/registerController/index.js

```diff
@@ -0,0 +1,51 @@
+const passport = require('passport');
+require('../../passportConfig')(passport);
+const User = require('../../models/user');
+const bcrypt = require('bcryptjs');
+const getDbConnection = require("../../dbConnection");
+
+function register(req, res) {
+    getDbConnection();
+    //Validate the request here before running query of DB
+    User.findOne({username: req.body.username}, async (err,document) => {
+        if(err) {
+            res.status(500).json({
+                success: false,
+                message: 'Database Failure',
+                info: ''
+            });
+        }
+        if(document) {
+            res.status(400).json({
+                success: false,
+                message: 'User Exists',
+                info: ''
+            });
+        }
+        if(!document) {
+            const hashedPassword = await bcrypt.hash(req.body.password, 10);
+            try {
+                const newUser = new User({
+                    username: req.body.username,
+                    password: hashedPassword,
+                    email: req.body.email || '',
+                    bio: req.body.bio || ''
+                })
+                await newUser.save();
+            } catch (error) {
+                res.status(500).json({
+                    success: false,
+                    message: 'Database Failure ' + error,
```

Comment:
Be cautious about returning the exact database error to the front-end - can be exposing too much information to hackers. Best to log the error on the backend server into a log file where it can be checked by someone with access to the server.